pkg/northbound/gnmi/set.go

```diff
@@ -208,10 +221,41 @@ func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
 		}
 		networkChanges[store.ConfigName(configName)] = changeID
 	}
-
 	return updateResults, networkChanges, nil
 }
 
+func listenForDeviceResponse(changes mapNetworkChanges, target string, name store.ConfigName) (bool, error) {
+	mgr := manager.GetManager()
+	respChan, ok := mgr.Dispatcher.GetResponseListener(topocache.ID(target))
+	if !ok {
+		log.Infof("Device %s not properly registered, not waiting for southbound confirmation", target)
+		return true, nil
+	}
+	response := <-respChan
+	switch eventType := response.EventType(); eventType {
+	case events.EventTypeAchievedSetConfig:
+		return true, nil
```

Comment:
does this mean you're going to return on the first successful device - shouldn't you wait for all of them?

Reply:
no, this is a per device process, meaning that this device achieved correctly the set, if the next one fails we will roll it back. I can achieve this because the set is sequential per target, see comment at line 165 fo set.go

Reply:
OK